src/data/db.ts

```diff
@@ -0,0 +1,202 @@
+import { Promise, Thenable } from "bluebird";
+import * as pgPromise from "pg-promise";
+import * as R from "ramda";
+
+import { database } from "../config";
+import { slugify } from "../app/helpers";
+
+// Limit the amount of debugging of SQL expressions
+const trimLogsSize: number = 200;
+const maxRecordsPerPage: number = 30;
+
+const pgpDefaultConfig = {
+  promiseLib: Promise,
+  query: (query: { query: string }): void => {
+    // Log all querys
+    console.log("[SQL   ]", R.take(trimLogsSize, query.query));
+  },
+  error: (err: Error, e: { query?: string }): void => {
+    // On error, please show me the SQL
+    if (e.query) {
+      console.error("[SQL   ]", R.take(trimLogsSize, e.query), err);
+    }
+  },
+};
+
+console.info(
+  "Connecting to the database:",
+  `${database.user}@${database.host}:${database.port}/${database.database}`
+);
+
+const pgp = pgPromise(pgpDefaultConfig);
+const db = pgp(database);
+
+interface GithubUsers {
+  id: number;
+  name: string;
+  login: string;
+  company: string;
+  blog: string;
+  email: string;
+  location: string;
+  bio: string;
+  hireable: boolean;
+  languages?: string[];
+}
+
+function findLanguagesBySlug(slugs: string[]) {
+  return db.manyOrNone(
+    "SELECT id, slug FROM programming_language WHERE slug IN ($1:csv);",
+    slugs
+  );
+}
+
+function getLanguageIds(languages: string[]) {
+  const languageSlugs = languages.map(slugify);
+
+  return findLanguagesBySlug(languageSlugs).then((existingLanguages) => {
+    const existingLanguagesSet = new Set(
+      existingLanguages.map((language) => language.slug)
+    );
+
+    const missingLanguages = languages.filter(
+      (language) => !existingLanguagesSet.has(slugify(language))
+    );
+
+    if (!missingLanguages.length)
```

Comment:
This does work, and it makes sense, but it's better typed if you use the strict check operator `===`.
For now, the `length` property is an integer, but we want it to work **only** for integers.
If it's `null` or any of its brothers, we want to treat it differently, but we don't need to cover those cases here.

Reply:
Done!